Assets/MixedRealityToolkit.SDK/Experimental/Dialog/Scripts/Dialog.cs

```diff
@@ -0,0 +1,179 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.Dialog
+{
+    public abstract class Dialog : MonoBehaviour
+    {
+        /// <summary>
+        /// It's used to keep track of the current state in which the Dialog is
+        /// </summary>
+        protected DialogState state = DialogState.Uninitialized;
+        public DialogState State
+        {
+            get => state;
+            set => state = value;
+        }
+
+        /// <summary>
+        /// Called after user has clicked a button and the dialog has finished closing
+        /// </summary>
+        public Action<DialogResult> OnClosed;
+
+        protected DialogResult result;
+        /// <summary>
+        /// Can be used to monitor result instead of events
+        /// </summary>        
+        public DialogResult Result
+        {
+            get => result;
+        }
```

Comment:
nit: `public DialogResult Result => result;`
If you don't have time, it's fine.